### PR TITLE
Session fixation fix

### PR DESCRIPTION
**Pull request by miteto**
# Session fixation fix

## Description
check for valid session ID before inserting it into session table

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [X] 2.8.x
- [X] 18.10.x
- [X] 19.04.x
- [X] 19.10.x (master)

### CODE REVIEW

File: www/include/core/login/processLogin.php [hunk 1]
```diff
@@ -99,29 +99,36 @@
...
-            if (isset($_GET["min"]) && $_GET["min"] == '1') {
-                $minimize = '&min=1';
-            }
-            if (isset($_GET["p"]) && $_GET["p"] != '') {
-                header('Location: ./main.php?p=' . $_GET["p"] . $minimize);
-            } else if (isset($centreon->user->default_page) && $centreon->user->default_page != '') {
-                header('Location: ./main.php?p=' . $centreon->user->default_page . $minimize);
+        // Check for session
+        $sessionId = session_id();
+        if (!preg_match('/^[a-zA-Z0-9,-]{26}$/', $sessionId)) {
```
Comment (sc979) [inline www/include/core/login/processLogin.php]: you check that the PHPSESSID is exactly 26 characters long.
but I can't find were you check that the PHPSESSID is the same as the one set by the session ? (eg: not modified by a user to use someone else identity)

File: www/include/core/login/processLogin.php [hunk 2]
```diff
@@ -99,29 +99,36 @@
...
+                $minimize = '';
+                if (isset($_GET["min"]) && $_GET["min"] == '1') {
+                    $minimize = '&min=1';
+                }
+                if (isset($_GET["p"]) && $_GET["p"] != '') {
+                    header('Location: ./main.php?p=' . $_GET["p"] . $minimize);
+                } else if (isset($centreon->user->default_page) && $centreon->user->default_page != '') {
+                    header('Location: ./main.php?p=' . $centreon->user->default_page . $minimize);
+                } else {
+                    header('Location: ./main.php');
```
Comment (sc979) [inline www/include/core/login/processLogin.php]: why using single quote here, when you use doubles line 128 ?

File: www/include/core/login/processLogin.php [hunk 3]
```diff
@@ -99,29 +99,36 @@
...
+        // Check for session
+        $sessionId = session_id();
+        if (!preg_match('/^[a-zA-Z0-9,-]{26}$/', $sessionId)) {
+            header("Location: ./index.php?disconnect=1");
+            $connect = false;
+        } else {
+            $query = "INSERT INTO `session` (`session_id` , `user_id` , `current_page` , `last_reload`, `ip_address`) " .
+                "VALUES (?, ?, ?, ?, ?)";
+            $dbResult = $pearDB->prepare($query);
+            $pearDB->execute(
```
Comment (sc979) [inline www/include/core/login/processLogin.php]:
```suggestion
            $dbResult->execute(
```

File: www/include/core/login/processLogin.php [hunk 4]
```diff
@@ -99,29 +99,36 @@
...
+                $dbResult,
```
Comment (sc979) [inline www/include/core/login/processLogin.php]:
```suggestion
```

File: www/include/core/login/processLogin.php [hunk 5]
```diff
@@ -99,29 +99,36 @@
...
+                if (isset($_GET["p"]) && $_GET["p"] != '') {
+                    header('Location: ./main.php?p=' . $_GET["p"] . $minimize);
+                } else if (isset($centreon->user->default_page) && $centreon->user->default_page != '') {
+                    header('Location: ./main.php?p=' . $centreon->user->default_page . $minimize);
+                } else {
+                    header('Location: ./main.php');
+                }
             } else {
-                header('Location: ./main.php');
+                header("Location: ./main.php");
```
Comment (sc979) [inline www/include/core/login/processLogin.php]:
```suggestion
                header('Location: ./main.php');
```

File: www/include/core/login/processLogin.php [hunk 6]
```diff
@@ -99,29 +99,36 @@
...
+                [$sessionId, $centreon->user->user_id, '1', time(), $_SERVER["REMOTE_ADDR"]]
+            );
+            if (!isset($_POST["submit"])) {
+                $minimize = '';
+                if (isset($_GET["min"]) && $_GET["min"] == '1') {
+                    $minimize = '&min=1';
+                }
+                if (isset($_GET["p"]) && $_GET["p"] != '') {
+                    header('Location: ./main.php?p=' . $_GET["p"] . $minimize);
+                } else if (isset($centreon->user->default_page) && $centreon->user->default_page != '') {
```
Comment (loiclau) [inline www/include/core/login/processLogin.php]:
```suggestion
                } elseif (isset($centreon->user->default_page) && $centreon->user->default_page != '') {
```

**Comment by sc979** [comment]
as the cookie is still vulnerable. I close this PR, which is replaced by this one : https://github.com/centreon/centreon/pull/7892

**Pull request closed without merging**